docker-images/kafka/scripts/zookeeper_run.sh

```diff
@@ -51,5 +53,34 @@ fi
 
 . ./set_kafka_gc_options.sh
 
+############ ZK Upgrade Start ############
+# ZK 3.4.x -> 3.5.x upgrade adds snapshot checks that can cause ZK pods with persistent storage to fail on startup
+# The section below disables the snapshot checks until the servers have created a snapshot.
+
+# If this is a 3.4.x server then place an indicator file in the data directory so this persists over multiple restarts
+flag_file=${ZOOKEEPER_DATA_DIR}/zk_upgrade_from_3_4
+if [[ $(ls libs | grep -Po 'zookeeper-\K3.4.\d+' | head -1) ]]; then
+    touch ${flag_file}
+fi
+
+# If this server is attached to a disk that was once used by a 3.4.x server then we may need to disable snapshot checks
+if [[ -f ${flag_file} ]]; then
```

Comment:
I do not understand why are you setting the `zookeeper.snapshot.trust.empty` option from two levels. Why is one not sufficient?

Reply:
Do you mean why I do the library version check and then check the flag file that produces instead of just doing the lib version check as the only if. The answer is that on a newly spun up 3.5.6 server attached to a previous 3.4.x disk the above check would fail and the snapshots would not be checked. This way the flag file will persist across the upgrade to let the 3.5.6 server know that it may need to skip the snapshot checks.

Reply:
If however, you mean why are we setting a CO env var and doing this, the answer is I somehow messed up a rebase and didn't remove the CO env var setting code. That will be removed and this script based check will be the only one.

Reply:
Wait ... I'm not sure that really helps. How will you know in CO what are you upgrading from when the version field is not set? I guess the only workaround is to detect the upgrade from something else.

Reply:
Another very good point

Reply:
Can we assume that if there is no version information set and you are upgrading to 0.16 that you must be coming from a pre-3.5.6 ZK install and therefore if no version number is set that the checks should be disable (until snapshots are present) by default?